yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/call/identity/DigitalIdentityService.java

```diff
@@ -146,8 +160,43 @@ public ShareSessionQrCode fetchShareQrCode(String sdkId, KeyPair keyPair, String
         }
     }
 
-    public Object fetchShareReceipt(String receiptId) {
-        return null;
+    public ReceiptDetails fetchShareReceipt(String sdkId, KeyPair keyPair, String receiptId)
+            throws DigitalIdentityException {
+        String receiptPath = pathFactory.createIdentitySessionReceiptRetrievalPath(receiptId);
+
+        ShareSessionReceipt receipt;
+        try {
+            SignedRequest request = createSignedRequest(sdkId, keyPair, receiptPath);
+            receipt = request.execute(ShareSessionReceipt.class);
+        } catch (Exception ex) {
+            throw new DigitalIdentityException(
+                    String.format("Error while fetching the share session QR code with ID '{%s}' ", receiptId),
+                    ex
+            );
+        }
```

Comment:
When doing assignments "wrapped" in a try/catch if makes for more readable code if the try/catch is extracted to a private method (so the try/catch always returns a value instead of assigning one), e.g.
```java
ShareSessionReceipt receipt = doFetchShareReceipt(sdkId, keyPair, receiptPath);
ShareSessionReceiptItemKey receiptKey = fetchShareReceiptKey(sdkId, keyPair, receipt);
```

Extracting the try/catch to a private method, removes the language "cruft" from the "main" logic making it easier to focus on the business logic (**what** is actually being done) instead of **how** it's being done).